engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java

```diff
@@ -417,7 +417,7 @@ boolean equalsInteropObjectWithMembers(
     }
   }
 
-  @Specialization(guards = {"isHostObject(selfHostObject)", "isHostObject(otherHostObject)"})
+  @Specialization(guards = {"isJavaObject(selfHostObject)", "isJavaObject(otherHostObject)"})
```

Comment:
I find this renaming a bit unintuitive. `JavaObject` can be both *host* and *guest* object (depending on whether the object was created in Espresso). No? Shouldn't we keep the old naming?

Reply:
Espresso runs as a guest language, and as such it produces  _"guest objects"_. Depending on whether we use Enso regular Java interop or Espresso interop the definition of `isJavaObject` changes - either it checks for host objects or it checks for objects coming from Espresso. That's why I was searching for some uniform name and came up with `isJavaObject`.

True, we don't want Java object from the interpreter, just Java object from regular or Espresso interop. So I agree the naming isn't perfect.